static/js/stream_edit.js

```diff
@@ -646,9 +646,14 @@ exports.initialize = function () {
     $("#subscriptions_table").on("click", ".sub_unsub_button", function (e) {
         var sub = get_sub_for_target(e.target);
         var stream_row = $(this).parent();
-        subs.sub_or_unsub(sub);
         var sub_settings = settings_for_sub(sub);
         var regular_sub_settings = sub_settings.find(".regular_subscription_settings");
+
+        // If checkmark is disabled(user is not allowed to subscribe to stream)
+        // then don't send ajax request, only open stream settings
+        if (!$(e.target).closest(".sub_unsub_button").hasClass("disabled")) {
+            subs.sub_or_unsub(sub);
```

Comment:
I have tried to implement this another way, i.e setting `pointer-event: none;` for `disabled` class, but that will also silence hover event on class.